File: Java/기본 수학2/베르트랑 공준.java
import java.io.*;
import java.util.StringTokenizer;

public class Main{
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken());
		
		while(N != 0) {
			int cnt = 0;
			
			for(int i = N + 1; i < 2*N +1; i++) {
				int tmp = 0;
				
				if(i > 1) {
					int t = (int)Math.sqrt(i);
					for(int j = 2; j < t + 1; j++) {
						if(i % j == 0) {
							tmp = -1;
							break;
						}
					}
				}
				
				if(tmp == 0) cnt++;
			}
			System.out.println(cnt);
			st = new StringTokenizer(br.readLine());
			N = Integer.parseInt(st.nextToken());
		}
    }
}

File: C++/기본 수학2/베르트랑 공준.cc
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    while(n != 0) {
        int cnt = 0;
        
        if(n == 1)
            cout << 1 << '\n';
        else {
            for(int i = n + 1; i <= 2*n; i++) {
                if(i % 2) {
                    for(int j = 3; j < (int)sqrt(i) + 1; j += 2) {
                        if(i % j == 0) {
                            cnt--;
                            break;
                        }
                    }
                    cnt++;
                }
            }
            cout << cnt << '\n';
        }
        
        cin >> n;
    }
    
    return 0;
}